From 95c600332f60dbdf3c6d886a149773a306a6e388 Mon Sep 17 00:00:00 2001
From: Han Chao <chan@windriver.com>
Date: Wed, 9 Apr 2014 09:57:11 +0800
Subject: [PATCH 6/6] grsec: add condition flag to disable GRSecurity.

grsec_tcb_lock should be only compiled when CONFIG_GRKERNSEC is enabled.

Signed-off-by: Han Chao <chan@windriver.com>
---
 fs/compat.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/compat.c b/fs/compat.c
index 40eb043..3e960a0 100644
--- a/fs/compat.c
+++ b/fs/compat.c
@@ -853,12 +853,14 @@ static int compat_fillonedir(void *__buf, const char *name, int namlen,
 		return -EOVERFLOW;
 	}
 
+#ifdef CONFIG_GRKERNSEC
 	read_lock(&grsec_tcb_lock);
 	if (!gr_acl_handle_filldir(buf->file, name, namlen, ino)) {
 		read_unlock(&grsec_tcb_lock);
 		return 0;
 	}
 	read_unlock(&grsec_tcb_lock);
+#endif
 
 	buf->result++;
 	dirent = buf->dirent;
-- 
1.7.9.5

